espflash/src/cli/monitor/mod.rs

```diff
@@ -0,0 +1,264 @@
+use std::{
+    io::{stdout, ErrorKind, Write},
+    time::Duration,
+};
+
+use crossterm::{
+    event::{poll, read, Event, KeyCode, KeyEvent, KeyModifiers},
+    style::{Color, Print, PrintStyledContent, Stylize},
+    terminal::{disable_raw_mode, enable_raw_mode},
+    QueueableCommand,
+};
+use lazy_static::lazy_static;
+use miette::{IntoDiagnostic, Result};
+use regex::Regex;
+
+use self::{line_endings::normalized, symbols::Symbols};
+use crate::{connection::reset_after_flash, interface::Interface};
+
+mod line_endings;
+mod symbols;
+
+// Pattern to much a function address in serial output.
+// All function addresses begin with '0x4', followed by exactly 7 hex digits.
+lazy_static! {
+    static ref RE_FN_ADDR: Regex = Regex::new(r"0x4[[:xdigit:]]{7}").unwrap();
```

Comment:
> All function addresses begin with '0x4

Not true for functions in IRAM and also sometimes the names of statics are interesting